librz/bin/p/bin_omf166.c

```diff
@@ -0,0 +1,428 @@
+// SPDX-FileCopyrightText: 2015-2019 ampotos <mercie_i@epitech.eu>
+// SPDX-FileCopyrightText: 2015-2019 pancake <pancake@nopcode.org>
+// SPDX-FileCopyrightText: 2025 Sergey Sharshunov <s.sharshunov@gmail.com>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_types.h>
+#include <rz_util.h>
+#include <rz_lib.h>
+#include <rz_bin.h>
+#include "omf/omf.h"
+
+// Modified from one in analysis_riscv
+// First arg is checked against all others
+#define is_any_n(...) _is_any_n(__VA_ARGS__, NULL)
+static bool _is_any_n(const char *str, size_t n, ...) {
+	char *cur;
+	va_list va;
+	va_start(va, n);
+	while (true) {
+		cur = va_arg(va, char *);
+		if (!cur) {
+			break;
+		}
+		if (!strncmp(str, cur, n)) {
+			va_end(va);
+			return true;
+		}
+	}
+	va_end(va);
+	return false;
+}
+
+static bool load_buffer(RzBinFile *bf, RzBinObject *obj, RzBuffer *b, Sdb *sdb) {
+	ut64 size;
+	const ut8 *buf = rz_buf_data(b, &size);
+	rz_return_val_if_fail(buf, false);
+
+	obj->bin_obj = rz_bin_internal_omf166_load(buf, size);
+	rz_return_val_if_fail(obj->bin_obj, false);
+	return true;
+}
+
+static void destroy(RzBinFile *bf) {
+	rz_bin_omf166_obj *omf_obj = (rz_bin_omf166_obj *)bf->o->bin_obj;
+	if (bf->o->lines) {
+		RzBinSourceLineInfo *lines = bf->o->lines;
+		size_t sc = lines->samples_count;
+		for (size_t i = 0; i < sc; i++) {
+			RzBinSourceLineSample *sample = &lines->samples[i];
+			RZ_FREE(sample->file);
+		}
+		RZ_FREE(lines->samples);
+	}
+	ht_up_free(omf_obj->ht_types);
+	rz_bin_free_all_omf166_obj(omf_obj);
+	RZ_FREE(omf_obj);
```

Comment:
you can just call `free` directly, as you are not going to reuse the pointers here.